code/obj/flock/structure/sentinel.dm

```diff
@@ -0,0 +1,104 @@
+//
+// # Sentinel structure,
+//
+#define NOT_CHARGED -1 //! The sentinel is without charge
+#define LOSING_CHARGE 0 //! The sentinel is losing charge
+#define CHARGING 1 //! The sentinel is gaining charge
+#define CHARGED 2 //! The sentinel is charged
+/obj/flock_structure/sentinel
+	name = "Glowing pylon"
+	desc = "A glowing pylon of sorts, faint sparks are jumping inside of it."
+	icon_state = "sentinel"
+	flock_id = "Sentinel"
+	health = 80
+	var/charge_status = NOT_CHARGED
+	/// 0-100 charge percent
+	var/charge = 0
+	var/powered = FALSE
+
+	event_handler_flags = USE_CANPASS
+	/// flockdrones can pass through this
+	passthrough = TRUE
+
+	usesgroups = TRUE
+
+	/// debug amount scale up if needed.
+	poweruse = 20
+
+/obj/flock_structure/sentinel/New(var/atom/location, var/datum/flock/F=null)
+	..(location, F)
+	src.filters = filter(type="rays", x=-0.2, y=6, size=1, color=rgb(255,255,255), offset=rand(1000), density=20, threshold=0.2, factor=1)
+	var/f = src.filters[length(src.filters)]
+	animate(f, size=((-(cos(180*(3/100))-1)/2)*32), time=5 MINUTES, easing=LINEAR_EASING, loop=-1, offset=f:offset + 100, flags=ANIMATION_PARALLEL)
+
+/obj/flock_structure/sentinel/building_specific_info()
+	return {"<span class='bold'>Status:</span> [charge_status == 1 ? "charging" : (charge_status == 2 ? "charged" : "idle")].
+	<br><span class='bold'>Charge Percentage:</span> [charge]%."}
+
+/obj/flock_structure/sentinel/process()
+	updatefilter()
+
+	if(!src.group)//if it dont exist it off
+		powered = FALSE
+	else if(src.group.powerbalance >= 0)//if it has atleast 0 or more free power, the poweruse is already calculated in the group
+		powered = TRUE
+	else//if there isnt enough juice
+		powered = FALSE
+
+	if(powered == 1)
+		switch(charge_status)
+			if(NOT_CHARGED)
+				charge_status = CHARGING//begin charging as there is energy available
+			if(LOSING_CHARGE)
+				charge_status = CHARGING//if its losing charge and suddenly theres energy available begin charging
+			if(CHARGING)
+				if(icon_state != "sentinelon") icon_state = "sentinelon"//forgive me
+				src.charge(5)
+			if(CHARGED)
+				var/mob/m = null
+				var/list/hit = list()
+				for(m in mobs)
+					if(IN_RANGE(m, src, 5) && !isflock(m) && src.flock?.isEnemy(m) && isturf(m.loc))
+						break//found target
+				if(!m) return//if no target stop
```

Comment:
what the fuck is this, this does literally nothing - unless it somehow sets the value of m as it iterates??

even then, you have the same thing apparently being done below which is just whack shit